compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -1236,6 +1236,37 @@ impl<'a> Parser<'a> {
 
     /// Parse an indexing expression `expr[...]`.
     fn parse_index_expr(&mut self, lo: Span, base: P<Expr>) -> PResult<'a, P<Expr>> {
+        if self.prev_token.is_ident()
```

Comment:
This needs to check `self.may_recover()` as well.

Reply:
Actually, I guess this probably doesn't matter, since this isn't syntactically ambiguous at this point (expr fragments in macros cannot be followed by `[`)

But it probably doesn't hurt to add it.